src/testing/tests.rs

```diff
@@ -1650,4 +1651,155 @@ pub mod generic_wallet_tests {
+        ));
+
+        // Check that `asset1` got updated, retaining it's mint info and persistence but attaining
+        // verified status.
+        {
+            let asset1_info = wallets[0].0.asset(asset1.code).await.unwrap();
+            assert!(asset1_info.verified);
+            assert!(!asset1_info.temporary);
+            assert_eq!(asset1_info.definition, asset1);
+            assert!(asset1_info.mint_info.is_some());
+        }
+        // Check that `asset2`, which was not present before, got imported as-is.
+        assert_eq!(
+            wallets[0].0.asset(asset2.code).await.unwrap(),
+            AssetInfo {
+                definition: asset2.clone(),
+                mint_info: None,
+                verified: true,
+                temporary: true,
+            }
+        );
+
+        // Check that temporary assets are not persisted, and that persisted assets are never
+        // verified.
+        {
+            let storage = &ledger.lock().await.storage[0];
+            let loaded = storage.lock().await.load().await.unwrap();
+            assert!(loaded.assets.iter().all(|asset| !asset.verified));
+            assert!(loaded.assets.contains(AssetCode::native()));
+            assert!(loaded.assets.contains(asset1.code));
+            assert!(!loaded.assets.contains(asset2.code));
+            assert_eq!(loaded.assets.len(), 2);
+        }
+
+        // Now import `asset2`, updating the existing verified asset with persistence and mint info.
+        wallets[0]
+            .0
+            .import_asset(AssetInfo::new(asset2.clone(), mint_info2.clone()))
+            .await
+            .unwrap();
+        assert_eq!(
+            wallets[0].0.asset(asset2.code).await.unwrap(),
+            AssetInfo {
+                definition: asset2.clone(),
+                mint_info: Some(mint_info2),
+                verified: true,
+                temporary: false,
+            }
+        );
+
+        // Check that `asset2` is now persisted, but still no assets in storage are verified.
+        {
+            let storage = &ledger.lock().await.storage[0];
+            let loaded = storage.lock().await.load().await.unwrap();
+            assert!(loaded.assets.iter().all(|asset| !asset.verified));
+            assert!(loaded.assets.contains(AssetCode::native()));
+            assert!(loaded.assets.contains(asset1.code));
+            assert!(loaded.assets.contains(asset2.code));
+            assert_eq!(loaded.assets.len(), 3);
+        }
+
+        // Finally, check that by loading the persisted, unverified information, and combining it
+        // with the verified information, we get back the current in-memory information (which was
+        // generated in a different order).
+        let loaded = {
+            let storage = &ledger.lock().await.storage[0];
+            let mut assets = storage.lock().await.load().await.unwrap().assets;
+            for asset in imported {
+                assets.insert(asset);
```

Comment:
Should we also add `asset2` to `assets` since it's also an imported asset (though not in the `verified_assets` before)?

Reply:
`asset2` is already in `imported`, so it does get added here.

The point of this is to simulate the workflow of mixing some verified and unverified information, persisting just the unverified information, then opening the wallet later and loading back in the verified information. So I want to add the verified assets exactly as they are in `imported`.

Reply:
This makes sense! Somehow I thought there were 3 assets, `asset0`, `1`, and `2`, so `asset2` wasn't in `imported`.